student/views.py

```diff
@@ -1,3 +1,23 @@
-# from django.shortcuts import render
+from django.shortcuts import render, HttpResponseRedirect
+from .models import Student
+from .forms import UpdateStudentAccountSettingsForm
 
-# Create your views here.
+
+def account_update_success(request):
+    is_student = Student.is_student(request.user.id)
+    return render(request, 'account_settings_update_success.html', {"is_student": is_student})
+
+
+def update_student_account_settings_view(request):
+    context = {}
+    try:
+        student_object = Student.objects.get(user_id=request.user.id)
+    except Student.DoesNotExist:
+        return render(request, "dead_end.html")
+    update_student_form = UpdateStudentAccountSettingsForm(request.POST or None, instance=student_object)
+    if update_student_form.is_valid():
+        update_student_form.save()
+        return HttpResponseRedirect("/account_update_success/")
+    context["form"] = update_student_form
+    context["is_student"] = Student.is_student(request.user.id)
```

Comment:
I'm not sure if that's the ideal way to do it, @ifireball WDYT?

Reply:
@guy9050 
I found a much better way to do it,
I added a related name to the link between user and Student, and user and Recruiter, this way in the HTML I can check directly if the logged in user is a student or a recruiter without having to provide anything from the outside, I'll update it in a few minutes so you can check it out.

Reply:
@guy9050  Could you give me more context? What are you asking me about exactly?

@guyy13 Not sure moving logic into the HTML template is a good idea, at the very least it makes to hard to test. I would look for ways of moving it out of there. In this example, and without knowing a lot about the details, I would say that my expectation would be to have completely different page templates for student and requiters with the view selecting between them. Or perhaps even having different views and routes for each.

Reply:
@ifireball 
we do have different views and pages for each, the discussion here is about linking them to the navigation bar, which has to contain different links depending on who is logged in.
I just pushed the changes, please take a look.

Reply:
I see this is no longer in the code, though I'm not sure how are you handling the navigation without this and I don't have the time to dive deeper right now.

When you have a set of context values that need to go into many pages in order to get embedded in shared UI items such as navigation bars, its usually a good idea to move them into a function that gets called from all the relevant views. That makes it easier to know which context items are relevant for the particular view and which are shared.

Reply:
@ifireball 
Ok, anyway I managed to get around this with a solution that allows you to check from within the template directly whether the logged in user is Student or a Recruiter.
I think it's simpler than creating a function like you suggested. Welcome to take a look when you have time.

Reply:
Ok, I see those "ifs" you have in the template. Here is the key question - how do you test that?

Reply:
@ifireball 
I guess I could recreate the same code in a pytest test or something?
How would I test it if I did it as you suggested (a function that contains the shared context)? I'm not sure I understand

Reply:
If you recreate the code in the test - you are not testing the actual code...

To make it testable - you need to move the logic out of the template - one way to do that is to write different included sub-templates for the parts that need to be different according to logic, and put the name of the part you need to include in the context. This allows you to write a test that ensures you include the right sub-template name in the context in various scenarios.

Reply:
@ifireball 
Ok, I'll try to do that later